src/OrbitGl/App.cpp

```diff
@@ -1706,7 +1713,18 @@ orbit_base::Future<ErrorMessageOr<std::filesystem::path>> OrbitApp::RetrieveModu
         });
   };
 
-  return UnwrapFuture(check_file_on_remote.Then(main_thread_executor_, std::move(download_file)));
+  Future<ErrorMessageOr<std::filesystem::path>> chained_result_future =
+      UnwrapFuture(check_file_on_remote.Then(main_thread_executor_, std::move(download_file)));
+
+  symbol_files_currently_downloading_.emplace(
+      module_file_path,
+      OrbitApp::ModuleLoadOperation{std::move(stop_source), chained_result_future});
+  chained_result_future.Then(main_thread_executor_,
+                             [this, module_file_path](const auto& /*result*/) {
```

Comment:
replacing `auto` by the type would make understanding easier, is it `ErrorMessageOr<void>& result`?

Reply:
It is ErrorMessageOr<std::filesystem::path>. I added it here and one other location